.github/workflows/publish-containers.yml

```
@@ -0,0 +1,113 @@
+    branches:
+      - main
+      - ci/**
+
+jobs:
+  docker:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      # Outputs the name of the repository (owner/repo)
+      - name: Get image name
+        id: image
+        run: |
+          IMAGE_NAME=$(echo ${{ github.repository }} | tr '[:upper:]' '[:lower:]')
+          echo ::set-output name=image_name::ghcr.io/${IMAGE_NAME}
+
+      # Tags stable versions as :latest
+      # Pre-releases, alphas, etc. as :snapshot
+      - name: Output image tags from git tag events
+        if: ${{ startsWith(github.ref, 'refs/tags/') }}
+        run: |
+          set -x
+          # Remove everything else than the tagged version
+          VERSION=${GITHUB_REF#refs/tags/}
+          # Tag as stable (:latest) if there is no letters in the version
+          # Otherwise, tag as preview (:snapshot)
+          if [[ $VERSION =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+            # Matches 1.0.0, 1.0.1, etc.
+            NAMED_VERSION="latest"
+          else
+            # 1.0.0a1, 1.0.0rc1, etc.
+            NAMED_VERSION="snapshot"
+          fi
+          TAGS=$"\
+          ${{ steps.image.outputs.image_name }}:${NAMED_VERSION},\
+          ${{ steps.image.outputs.image_name }}:${VERSION}\
+          "
+          echo "
+          CONTAINER_TAGS=${TAGS}
+          NAMED_VERSION=${NAMED_VERSION}
+          " >> "${GITHUB_ENV}"
+
+      - name: Output image tags from git branch events
+        if: ${{ startsWith(github.ref, 'refs/heads/') }}
+        run: |
+          set -x
+          # Version name is the branch name, slashes are substitued by dashes
+          CLEAN_BRANCH_NAME=$(echo ${GITHUB_REF#refs/heads/} | tr / -)
+          # Target with the branch name as the tag, e.g. unstable-main
+          TAGS=$"\
+          ${{ steps.image.outputs.image_name }}:unstable-${CLEAN_BRANCH_NAME}
+          "
+          # Set version name for open-containers version label as:
+          # <branch>-<drift>-g<6-digit-hash>
+          NAMED_VERSION=$(git describe --all --long | tr -d $'\n')
+          echo "
+          CONTAINER_TAGS=${TAGS}
+          NAMED_VERSION=${NAMED_VERSION}
+          " >> "${GITHUB_ENV}"
+
+      - name: Set up Docker QEMU
+        uses: docker/setup-qemu-action@v1
+        with:
+          platforms: all
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-${{ hashFiles('requirements_dev.txt') }}
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ secrets.GHCR_USERNAME }}
+          password: ${{ secrets.GHCR_PASSWORD }}
```

That's a really outdated and insecure way of authenticating to GHCR 
```suggestion
          username: ${{ github.actor }}
          password: ${{ secrets.GITHUB_TOKEN }}
```

And add to the job definitions:
```
    permissions:
      contents: read
      packages: write
```